From: Haren Myneni <haren@us.ibm.com>                                                                                                                                
Subject: [PATHC] Trivial fix to set the proper timeout value for kdump                                                                                               

Bug 149725 - LTC21422-Using some invalid timeout value for waiting to get all CPUs


The panic CPU is waiting forever due to some large timeout value if some
CPU is not responding to an IPI.
This patch will fixes this issue -  The maximum waiting period will be
10 seconds and does the kdump boot.

Signed-off-by: Olaf Hering <olh@suse.de>

--- 2616-rc2.orig/arch/powerpc/kernel/crash.c	2006-02-07 15:20:26.000000000 -0800
+++ 2616-rc2/arch/powerpc/kernel/crash.c	2006-02-07 15:20:54.000000000 -0800
@@ -134,8 +134,10 @@ static void crash_kexec_prepare_cpus(voi
 	 * the crash CPU will send an IPI and wait for other CPUs to
 	 * respond. If not, proceed the kexec boot even though we failed to
 	 * capture other CPU states.
+	 * Delay of at least 10 seconds.
 	 */
-	msecs = 1000000;
+	printk(KERN_ALERT "Sending IPI to other cpus...\n");
+	msecs = 10000;
 	while ((atomic_read(&waiting_for_crash_ipi) > 0) && (--msecs > 0)) {
 		barrier();
 		mdelay(1);
